lib/util/splice.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Plain old regular Array#splice may cause stackoverflow if we have too many items
+ */
+function chunkedSplice(array, start, deleteCount, items) {
+  var chunkSize = 10000
+  var splice = [].splice
+
+  // Make deleteCount >= 0
+  deleteCount = Math.max(deleteCount, 0)
+  // Make start between 0 and array.length (included)
+  start = Math.min(start, array.length)
+  if (start < 0) {
+    start = Math.max(array.length + start, 0)
+  }
```

Comment:
(I believe the first case works)

```suggestion
  if (start < 0) {
    start = -start > array.length ? 0 : array.length + start
  } else {
      start = start > array.length ? array.length : start
  }
```